code/modules/admin/topic.dm

```diff
@@ -1586,58 +1531,58 @@ Status: [status ? status : "Unknown"] | Damage: [health ? health : "None"]
 		if(!istype(H))
 			return
 
-		var/dresscode = input("Please select an outfit.", "Select Equipment") as null|anything in list("{Naked}", "{Job}", "{Custom}")
-		if(!dresscode)
+		usr.client.holder.select_equipment(H)
+
+
+	else if(href_list["sleep"])
+		if(!check_rights(R_ADMIN))
```

Comment:
```suggestion
		if(!check_rights(R_ADMIN))
			log_admin("[key_name(usr)] tried to use the admin panel without authorization.")
			message_admins("[ADMIN_TPMONTY(usr)] tried to use the admin panel without authorization.")
```

Reply:
This could be a macro.

Reply:
..... it's already checked at the very first line of topic.dm```/datum/admins/Topic(href, href_list)
	. = ..()

	if(usr.client != owner || !check_rights(NONE))
		log_admin("[key_name(usr)] tried to use the admin panel without authorization.")
		message_admins("[ADMIN_TPMONTY(usr)] tried to use the admin panel without authorization.")
		return
```

Reply:
You should also read up on the purpose of check_rights and what it does by default

Reply:
Yep, and it's being checked again here.

Reply:
As it should since this is different